src/apps/btc/confirm_locktime_rbf.c

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Shift Cryptosecurity AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "confirm_locktime_rbf.h"
+
+#include <workflow/blocking.h>
+#include <workflow/status.h>
+
+#include <ui/components/confirm_transaction.h>
+#include <ui/screen_stack.h>
+#include <util.h>
+#include <workflow/confirm.h>
+
+bool confirm_locktime_rbf(uint32_t locktime, enum rbf_flag rbf)
+{
+    char formatted_locktime_rbf[100] = {0};
+    const char* locktime_text = "Locktime on block:";
+    const char* rbf_text;
+    if (rbf == rbf_on) {
+        rbf_text = "Transaction is RBF";
+    } else if (rbf == rbf_off) {
+        rbf_text = "Transaction is not RBF";
+    } else {
+        rbf_text = "";
+    }
+    if (locktime == 0 && rbf == rbf_disabled) {
```

Comment:
it's awkward that part of the condition to show a dialog is at call site (`if (_locktime_applies || _rbf == rbf_on) {`) and part of it is here. please move both to a single place, so it's easier to follow the logic.

Reply:
Removed the check in confirm_locktime_rbf, added a comment toe the function explaining what is required from the input.